src/sshfs_mount/sftp_server.cpp

```diff
@@ -194,15 +194,13 @@ auto validate_path(const std::string& source_path, const std::string& current_pa
 
 void check_sshfs_status(mp::SSHSession& session, mp::SSHProcess& sshfs_process)
 {
-    try
+    if (sshfs_process.exit_recognized(250ms))
     {
-        if (sshfs_process.exit_code(250ms) != 0)
+        // This `if` is artificial and should not really be here. However there is a complex arrangement of Sftp and
+        // SshfsMount tests depending on this.
+        if (sshfs_process.exit_code(250ms) != 0) // TODO remove
```

Comment:
I'm fine with this and understand the conundrum.  We sometimes have to add little quirks in the main code for testing- this is just one of those cases.